### DISCUSSION TITLE
DxCore with Visual code

### DISCUSSION THREAD

**Discussion by sshwarts** [SpenceKonde/DxCore#153] [Q&A]
Spence:
This might be a VS Code problem but I thought I'd tell you (or anyone else who has this problem). I could not specify the "SerialUPDI with 4.7k resistor or diode (230400 baud)" programmer in VS Code. I had to manually set it in the arduino.json file. When trying to set the programmer in VS menu, only the SerialUPDI with 4.7k resistor or diode (Max. speed for Dx, 345600 baud and SerialUPDI with 4.7k resistor or diode (Max. Dx speed for HT43B534 adapters choices showed up.

**Comment by SpenceKonde** [comment]
So the slow one ALSO doesn't show up?

I don't see what it would object to though...
serialupdi230k.name=SerialUPDI with 4.7k resistor or diode (230400 baud)
serialupdi230k.protocol=uart
serialupdi230k.program.extra_params=-u {serial.port} -b 230400
serialupdi230k.program.protocol=uart
serialupdi230k.program.tool=serialupdi
serialupdi230k.bootloader.tool=serialupdi

serialupdi345k.name=SerialUPDI with 4.7k resistor or diode (Max. speed for
Dx, 345600 baud)
serialupdi345k.protocol=uart
serialupdi345k.program.extra_params=-u {serial.port} -b 345600
serialupdi345k.program.protocol=uart
serialupdi345k.program.tool=serialupdi
serialupdi345k.bootloader.tool=serialupdi

serialupdi57k.name=SerialUPDI with 4.7k resistor or diode (SLOW - 57600
baud)
serialupdi57k.protocol=uart
serialupdi57k.program.extra_params=-u {serial.port} -b 57600
serialupdi57k.program.protocol=uart
serialupdi57k.program.tool=serialupdi
serialupdi57k.bootloader.tool=serialupdi

serialupdiHT42.name=SerialUPDI with 4.7k resistor or diode (Max. Dx speed
for HT43B534 adapters only, 460800 baud, chunked)
serialupdiHT42.protocol=uart
serialupdiHT42.program.extra_params=-u {serial.port} -b 460800 --blocksize
32
serialupdiHT42.program.protocol=uart
serialupdiHT42.program.tool=serialupdi
serialupdiHT42.bootloader.tool=serialupdi

Say, do you also have megaTinyCore installed? It strikes me that one of the
few things those two offenders have in common,,,,, is that they also exist
on megaTinyCore....

On Thu, Oct 7, 2021 at 2:01 PM Scott Shwarts ***@***.***>
wrote:

> Spence:
> This might be a VS Code problem but I thought I'd tell you (or anyone else
> who has this problem). I could not specify the "SerialUPDI with 4.7k
> resistor or diode (230400 baud)" programmer in VS Code. I had to manually
> set it in the arduino.json file. When trying to set the programmer in VS
> menu, only the SerialUPDI with 4.7k resistor or diode (Max. speed for Dx,
> 345600 baud and SerialUPDI with 4.7k resistor or diode (Max. Dx speed for
> HT43B534 adapters choices showed up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/DxCore/discussions/153>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW4OKHAFTCZ4RI7PAJTUFXN6ZANCNFSM5FR3NBSQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Comment by sshwarts** [comment]
Indeed I do have both installed. Maybe VS Code goes through all programmers.txt and gets offended somehow? I'll try to remove megaTinyCore and see.

**Comment by nabelekt** [comment]
I have just DxCore installed, and I am able to access all four SerialUPDI programmers in VS Code.
